tests/inst/src/rpmostree.rs

```diff
@@ -25,9 +24,29 @@ pub(crate) struct Deployment {
 }
 
 pub(crate) fn query_status() -> Result<Status> {
-    let cmd = Command::new("rpm-ostree")
-        .args(&["status", "--json"])
-        .stdout(Stdio::piped())
-        .spawn()?;
-    Ok(serde_json::from_reader(cmd.stdout.unwrap())?)
+    // Retry on temporary activation failures, see
+    // https://github.com/coreos/rpm-ostree/issues/2531
+    let pause = std::time::Duration::from_secs(1);
+    let mut retries = 0;
+    let cmd_res = loop {
+        retries += 1;
+        let res = Command::new("rpm-ostree")
+            .args(&["status", "--json"])
+            .output()
+            .context("failed to spawn 'rpm-ostree status'")?;
+
+        if res.status.success() || retries >= 10 {
+            break res;
+        }
+        std::thread::sleep(pause);
+    };
+
+    if !cmd_res.status.success() {
+        anyhow::bail!(
+            "running 'rpm-ostree status' failed: {}",
+            String::from_utf8_lossy(&cmd_res.stderr)
```

Comment:
As a general rule...I think it's better to just redirect child process stderr to our stderr rather than capturing it, because the output can be arbitrarily large.